### PR TITLE
Fix consecutive thoughts removal on ride demolish

### PR DESCRIPTION
**Pull request by TyPR124**
Previously, if a peep had two consecutive thoughts about the same ride, only the first would be removed, leaving the 2nd thought about a non-existing ride.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Is there a bug report for this?

**Comment by TyPR124** [comment]
There is not, I just found this error while looking through the code. I did confirm on my client that the bug can happen, though.

**Comment by duncanspumpkin** [comment]
Can you add an entry to the changelog for it. I'll check and see if this happens in vanilla.

**Comment by duncanspumpkin** [comment]
No this was a bug introduced by us during implementation. Just add to the changelog stating it is fixed.

File: src/openrct2/actions/RideDemolishAction.hpp [hunk 1]
```diff
@@ -176,6 +176,7 @@ struct RideDemolishAction : public GameActionBase<GAME_COMMAND_DEMOLISH_RIDE, Ga
                     // Clear top thought, push others up
                     memmove(&peep->thoughts[i], &peep->thoughts[i + 1], sizeof(rct_peep_thought)*(PEEP_MAX_THOUGHTS - i - 1));
                     peep->thoughts[PEEP_MAX_THOUGHTS - 1].type = PEEP_THOUGHT_TYPE_NONE;
+                    i--; //Next iteration, check the new thought at this index
```
Comment (Gymnasiast) [inline src/openrct2/actions/RideDemolishAction.hpp]: Could you move the comment above the instruction like the other?

**Comment by duncanspumpkin** [comment]
I also dont see your name in the contributors list so add youself in there.

File: distribution/changelog.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-0.1.2 (in development)
+﻿0.1.2 (in development)
```
Comment (janisozaur) [inline distribution/changelog.txt]: Please revert this change.

**Comment by janisozaur** [comment]
This looks like it needs network version change.

File: distribution/changelog.txt (hunk 2 above)
Comment (AaronVanGeffen) [reply distribution/changelog.txt]: (It looks like your editor inserted a byte order mark at the start of the file.)

**Comment by TyPR124** [comment]
I'm not sure if all I need to do to change network version is change that line. Let me know if I need to change anything else.

**Comment by janisozaur** [comment]
> I'm not sure if all I need to do to change network version is change that line.

That's all that's needed.
